data/de/db-hafas-mgate.json

```diff
@@ -0,0 +1,59 @@
+{
+  "name": "Deutsche Bahn (DB)",
+  "type": "hafas_mgate_deutschebahn",
+  "supportedLanguages": [
+    "en",
+    "de",
+    "fr",
+    "es"
+  ],
+  "options": {
+    "auth": {
+      "aid": "n91dB8Z77MLdoR0K"
+    },
+    "checksumSalt": "6264493855566A34304B356676787766",
+    "client": {
+      "id": "DB",
+      "name": "DB Navigator",
+      "type": "AND",
+      "v": 19040000
+    },
+    "endpoint": "https://reiseauskunft.bahn.de/bin/mgate.exe",
+    "ext": "DB.R19.12.a",
+    "lineModeMap": {
+      "1": "Long-distance train (ICE)",
+      "128": "Metro",
+      "16": "Rapid Transit (S)",
+      "2": "Long-Distance Train (IC/EC)",
+      "256": "Tram",
+      "32": "Bus",
+      "4": "Local Train (D)",
+      "512": "Taxi",
+      "64": "Ferry",
+      "8": "Local Train (RE/RB)"
+    },
+    "locationIdentifierType": "db",
```

Comment:
What is the idea behind this? @vkrause That the IDs returned by the endpoint are DB-style IBNRs?

Reply:
KPublictTransport has multiple id "namespaces" per location. This can be endpoint-specific ones that have no meaning outside (the default), proprietary ones that are shared between two or more endpoints (BVG/VBB are such an example), or standard ones (IBNR, UIC, IFOPT, etc). This is useful for merging data from different sources (different backends, OSM, Wikidata, etc).

To support this we have the following settings:
- `locationIdentifierType` defines the id namespace. This is optional for proprietary id spaces not used anywhere else.
- For many Hafas-based endpoints there is the problem that they use an IBNR or UIC code for stations having one of those, but a proprietary numeric scheme for everything else. The `standardLocationIdentifierType` and `standardLocationIdentifierCountries` options address this, the list of covered UIC country codes is needed to reliably distinguish IBNR/UIC codes from other numeric values.

This is obviously very specific to what KPublicTransport does, not particularly elegant or generic, and for most existing users probably irrelevant. I could imagine something like this to be relevant for your merging work though?

Reply:
> This is obviously very specific to what KPublicTransport does, not particularly elegant or generic, and for most existing users probably irrelevant. I could imagine something like this to be relevant for your merging work though?

Yes, highly relevant for my merging work! In fact, there are several projects (in the European community) that try to cross-reference public transport "things" in some way.